test/cpp/tensorexpr/test_expr.cpp

```diff
@@ -164,6 +164,24 @@ void testExprDoubleTest() {
   ASSERT_EQ(eval.value<double>(), 2 + (3 * 3 + 4));
 }
 
+void testExprDisallowBoolArithmetic() {
+  KernelScope kernel_scope;
+  VarHandle x("x", kBool);
+  VarHandle y("y", kBool);
+  std::string error{"arithmetic binary operations on Bool not supported"};
+  ASSERT_THROWS_WITH((x + y), error);
+  ASSERT_THROWS_WITH((x - y), error);
```

Comment:
If I understand, you're basically saying bool arithmetic within the TE IR should be disallowed?  Sounds good to me.

Reply:
@bertmaher Yes, which addresses the testing debt introduced by https://github.com/pytorch/pytorch/pull/44659. By outright disallowing them, it's one less thing to worry about and write tests for.